cmd/occollector/app/builder/builder.go

```diff
@@ -106,6 +107,37 @@ type OpenCensusReceiverCfg struct {
 
 	// TLSCredentials is a (cert_file, key_file) configuration.
 	TLSCredentials *config.TLSCredentials `mapstructure:"tls_credentials"`
+
+	// Keepalive anchor for all the settings related to keepalive.
+	Keepalive *serverParametersAndEnforcementPolicy `mapstructure:"keepalive,omitempty"`
+
+	// MaxRecvMsgSizeMiB sets the maximum size (in MiB) of messages accepted by the server.
+	MaxRecvMsgSizeMiB uint64 `mapstructure:"max-recv-msg-size-mib"`
+
+	// MaxConcurrentStreams sets the limit on the number of concurrent streams to each ServerTransport.
+	MaxConcurrentStreams uint32 `mapstructure:"max-concurrent-streams"`
+}
+
+type serverParametersAndEnforcementPolicy struct {
+	ServerParameters  *keepaliveServerParameters  `mapstructure:"server-parameters,omitempty"`
+	EnforcementPolicy *keepaliveEnforcementPolicy `mapstructure:"enforcement-policy,omitempty"`
+}
+
+// keepaliveServerParameters allow configuration of the keepalive.ServerParameters.
+// See https://godoc.org/google.golang.org/grpc/keepalive#ServerParameters for details.
+type keepaliveServerParameters struct {
```

Comment:
Do we have a use case for allowing end user to configure all these options? I would be cautious and only expose what end users really need to be able to change. I am worried about proliferation of configuration options.

Reply:
I have all of those in use in different scenarios, granted that many users may do fine without them but we need to make these levers available.